### DISCUSSION TITLE
VVC Encoder

### DISCUSSION THREAD

**Discussion by tariqazmy** [fraunhoferhhi/vvenc#20] [General] [answered]
Hi,

Newbie question. What is the difference between the vvenc here and the one from VTM repository? 

Thanks!

**Comment by 1480c1** [comment]
The one in the VTM repo is considered a reference one and vvenc can be considered one implementation based on the VTM one

**Reply by tariqazmy** [reply]
Thanks for your reply. So basically VTM is more of an reference where you can experiment with different parameters and choose different modes (intra, random etc.) based of their config files. And VVenc is simplified and complete version with some fixed parameters and run without specifying any config file?

**Comment by adamjw24** [comment, marked as answer]
So the VTM is the reference software which has a very thorough implementation of all VVC tools and serves as a kind of proof-of-concept. It was designed to make it easy to experiment on with many implementation aspects being very flexible to facilitate experimentation of various tools and their variants. Because runtime is not an issue for VTM, and it should showcase the full potential of VVC, many of the search algorithm do a full, or at least a very thorough, search of the VVC search space.

On the other hand it, VVenC, while based on the VTM, has be refactored a lot. We stripped down a lot of functionality and implemented some aspects and tools from scratch. We experimented with various tools and were able to improve implementation of many tools. On top of that, we added alternative search algorithms to perform less extensive search, but at much faster runtimes. We tested the tools against the speed-up options against other high-level settings to define the optimal presets for various need (ranging from VTM gains at half its time to the "faster" preset that only provides a bit of gains vs HM, but is much faster).
On top of that we implemented more practical encoder aspects, that are not required for a reference software, like single and two-pass rate-control, multi-threading, and subjective quality optimization based on XPSNR.
To achieve all of this, we sacrificed some of the functionality, like multi-layer or specific RGB-content tools. While they might come in the future they are not a priority for now.

Hope this helps. So in short, VVenC is to VTM what x265 is to HM. Just an alternative implementation with different goals, even if based on the same roots.

**Reply by tariqazmy** [reply]
Thank you for brief answer.
